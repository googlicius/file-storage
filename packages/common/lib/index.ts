export function common() {
  // TODO
  return 'this is common';
}
