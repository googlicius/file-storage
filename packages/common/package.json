{
  "name": "@file-storage/common",
  "version": "1.2.7",
  "description": "Common utilities, types, interfaces,... for `file-storage`.",
  "author": "Dang Nguyen <haidang009@gmail.com>",
  "homepage": "",
  "license": "MIT",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "files": [
    "dist"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/googlicius/file-storage.git"
  },
  "publishConfig": {
    "registry": "https://registry.npmjs.org",
    "access": "public"
  },
  "scripts": {
    "build": "yarn clean && yarn compile",
    "clean": "rimraf -rf ./dist",
    "compile": "tsc -p tsconfig.build.json"
  },
  "dependencies": {
    "sharp": "^0.28.3"
  },
  "devDependencies": {
    "@types/sharp": "^0.28.5",
    "rimraf": "~3.0.2",
    "typescript": "~4.3.5"
  }
}
