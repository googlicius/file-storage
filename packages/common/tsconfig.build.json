{
  "extends": "../../tsconfig.build.json",
  "compilerOptions": {
    "outDir": "./dist"
  },
  "include": ["src/**/*.ts"],
  "exclude": ["**/*.spec.ts"]
}
