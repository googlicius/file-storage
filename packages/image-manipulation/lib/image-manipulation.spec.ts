import fs from 'fs';
import Storage from '@file-storage/core';
import { getRootCwd } from '@file-storage/common';
import ImageManipulation from '@file-storage/image-manipulation';

describe('Image Manipulation', () => {
  beforeAll(() => {
    Storage.config({
      plugins: [ImageManipulation],
    });
  });

  it('should generates 2 custom breakpoints: size1, size2', () => {
    const imageFileStream = fs.createReadStream(
      getRootCwd() + '/test/support/images/photo-1000x750.jpeg',
    );

    ImageManipulation.config({
      breakpoints: {
        size1: 400,
        size2: 600,
      },
    });

    return expect(Storage.put(imageFileStream, 'bird.jpeg')).resolves.toMatchSnapshot();
  });

  it('should generates custom thumbnail size', () => {
    const imageFileStream = fs.createReadStream(
      getRootCwd() + '/test/support/images/photo-1000x750.jpeg',
    );

    ImageManipulation.config({
      breakpoints: null,
      thumbnailResizeOptions: {
        width: 333,
        height: 222,
        fit: 'contain',
      },
    });

    return expect(Storage.put(imageFileStream, 'bird.jpeg')).resolves.toMatchSnapshot();
  });

  it('should have no responsive formats', () => {
    const imageFileStream = fs.createReadStream(
      getRootCwd() + '/test/support/images/photo-1000x750.jpeg',
    );

    ImageManipulation.config({
      breakpoints: null,
    });

    return expect(Storage.put(imageFileStream, 'bird.jpeg')).resolves.toMatchSnapshot();
  });

  it('should have no thumbnail', () => {
    const imageFileStream = fs.createReadStream(
      getRootCwd() + '/test/support/images/photo-1000x750.jpeg',
    );

    ImageManipulation.config({
      thumbnailResizeOptions: null,
    });

    return expect(Storage.put(imageFileStream, 'bird.jpeg')).resolves.toMatchSnapshot();
  });

  it('should not generate image sizes when uploading a file that not an image', () => {
    const fileStream = fs.createReadStream(getRootCwd() + '/test/support/files/test-file.txt');

    return expect(Storage.put(fileStream, 'test-file.txt')).resolves.toMatchSnapshot();
  });
});
