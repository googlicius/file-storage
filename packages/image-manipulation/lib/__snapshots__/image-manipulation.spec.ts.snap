// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Manipulation should generates 2 custom breakpoints: size1, size2 1`] = `
Object {
  "formats": Object {
    "size1": Object {
      "ext": "jpeg",
      "hash": null,
      "height": 300,
      "mime": "jpeg",
      "name": "size1_bird.jpeg",
      "path": "size1_bird.jpeg",
      "size": 35.91,
      "width": 400,
    },
    "size2": Object {
      "ext": "jpeg",
      "hash": null,
      "height": 450,
      "mime": "jpeg",
      "name": "size2_bird.jpeg",
      "path": "size2_bird.jpeg",
      "size": 71.02,
      "width": 600,
    },
    "thumbnail": Object {
      "ext": "jpeg",
      "hash": null,
      "height": 156,
      "mime": "jpeg",
      "name": "thumbnail_bird.jpeg",
      "path": "thumbnail_bird.jpeg",
      "size": 15.9,
      "width": 208,
    },
  },
  "message": "Uploading success!",
  "name": "bird.jpeg",
  "path": "bird.jpeg",
  "success": true,
}
`;

exports[`Image Manipulation should generates custom thumbnail size 1`] = `
Object {
  "formats": Object {
    "thumbnail": Object {
      "ext": "jpeg",
      "hash": null,
      "height": 222,
      "mime": "jpeg",
      "name": "thumbnail_bird.jpeg",
      "path": "thumbnail_bird.jpeg",
      "size": 24.8,
      "width": 333,
    },
  },
  "message": "Uploading success!",
  "name": "bird.jpeg",
  "path": "bird.jpeg",
  "success": true,
}
`;

exports[`Image Manipulation should have no responsive formats 1`] = `
Object {
  "formats": Object {
    "thumbnail": Object {
      "ext": "jpeg",
      "hash": null,
      "height": 156,
      "mime": "jpeg",
      "name": "thumbnail_bird.jpeg",
      "path": "thumbnail_bird.jpeg",
      "size": 15.9,
      "width": 208,
    },
  },
  "message": "Uploading success!",
  "name": "bird.jpeg",
  "path": "bird.jpeg",
  "success": true,
}
`;

exports[`Image Manipulation should have no thumbnail 1`] = `
Object {
  "formats": Object {
    "large": Object {
      "ext": "jpeg",
      "hash": null,
      "height": 750,
      "mime": "jpeg",
      "name": "large_bird.jpeg",
      "path": "large_bird.jpeg",
      "size": 184.49,
      "width": 1000,
    },
    "medium": Object {
      "ext": "jpeg",
      "hash": null,
      "height": 562,
      "mime": "jpeg",
      "name": "medium_bird.jpeg",
      "path": "medium_bird.jpeg",
      "size": 105.12,
      "width": 750,
    },
    "small": Object {
      "ext": "jpeg",
      "hash": null,
      "height": 375,
      "mime": "jpeg",
      "name": "small_bird.jpeg",
      "path": "small_bird.jpeg",
      "size": 52.33,
      "width": 500,
    },
  },
  "message": "Uploading success!",
  "name": "bird.jpeg",
  "path": "bird.jpeg",
  "success": true,
}
`;

exports[`Image Manipulation should not generate image sizes when uploading a file that not an image 1`] = `
Object {
  "formats": undefined,
  "message": "Uploading success!",
  "name": "test-file.txt",
  "path": "test-file.txt",
  "success": true,
}
`;
